storm-client/src/jvm/org/apache/storm/windowing/TimeTriggerPolicy.java

```diff
@@ -62,7 +62,9 @@ public void reset() {
 
     @Override
     public void start() {
-        executorFuture = executor.scheduleAtFixedRate(newTriggerTask(), duration, duration, TimeUnit.MILLISECONDS);
+        // initial delay is slightly less than the duration so that the initial tuples wont't expire due to time drift
+        long initialDelay = duration - Math.min((long) (duration * .1), 10);
+        executorFuture = executor.scheduleAtFixedRate(newTriggerTask(), initialDelay, duration, TimeUnit.MILLISECONDS);
```

Comment:
Also since this is likely going to rely on in-bolt state to avoid expiring tuples before the user sees them, I think we should add to the documentation that this set of policies don't guarantee at-least-once processing.